File: readbin.c
#include <stdio.h>
#include <ctype.h>

void printbin(int numero){
    int bit;
    for(bit = 128; bit > 0; bit >>= 1){
        if((numero & bit) == bit){
            printf("1");    
        }
        else{
            printf("0");    
        }
    }
}

int main(int argc, char* argv[]){
    int contador;
    char puntero[(1024*10)];
    int bytes;
    int bytesmaximo;
    int pesolectura;
    int pesoescritura;
    FILE *archivodebytes;
    bytes = 1;
    bytesmaximo = (1024*10);
    if(argc<=1){
        return 0; 
    }
    archivodebytes = fopen(argv[1],"ab+");
    pesolectura = fread(puntero, bytes, bytesmaximo, archivodebytes);
    printf("\nINICIA EL ANALISIS...\n\n");
    for(contador = 0; contador < (pesolectura / sizeof(unsigned char)); contador++){
            printf(" Bin[%d]=", contador); 
            printbin((int)((unsigned char)puntero[contador])); //BIN
            printf("  Hex[%d]=0X%02x ", contador, (int)((unsigned char)puntero[contador])); //HEX
            printf(" ASCII[%d]=%d ", contador, (int)puntero[contador]); // ASCII

            if((int)puntero[contador]<=-100){
                printf("   Char[%d]=", contador); 
            } else if((int)puntero[contador]<=-10){
                printf("    Char[%d]=", contador); 

            } else if((int)puntero[contador]<0){
                printf("     Char[%d]=", contador);

            } else if((int)puntero[contador]>=100){
                printf("    Char[%d]=", contador); 
            } else if((int)puntero[contador]>=10) {
                printf("     Char[%d]=", contador); 
            } else{
                printf("      Char[%d]=", contador); 
            }
            printf("%c",(char)puntero[contador]); //CHAR
            printf("\n");
    }
    
    printf("\n\nSTRING: \n\n");
    for(contador = 0; contador < (pesolectura / sizeof(unsigned char)); contador++){

        if ( ((int)puntero[contador]) != 0 ){
            if (isalpha((char)puntero[contador]) != 0){
                printf("%c",(char)puntero[contador]);
            }
            else if (isdigit((char)puntero[contador]) != 0){
                printf("%c",(char)puntero[contador]);
            } else {
                printf(" ");
            }
        } 

    }
    fclose(archivodebytes);

    printf("\n\nFIN DEL ANALISIS...\n\n");
    return 0; 
}

File: readbin.go
package main

import (
        "fmt"
        "log"
        "os"
        "unicode"
)

func main(){
    if len(os.Args) == 1 {
        return
    }
    fmt.Print("\nINICIA EL ANALISIS...\n\n");
    file, err := os.Open(os.Args[1])
    if err != nil {
        log.Fatal(err)
    }
    fi, err := file.Stat()
    if err != nil {
        log.Fatal(err)
    }
    data := make([]byte, fi.Size())
    count, err := file.Read(data)
    if err != nil {
        log.Fatal(err)
    }
    for i := 0; i < count; i++ {
	fmt.Print("Bin[",i,"]=",fmt.Sprintf("%08b", data[i]))
	fmt.Print("  Hex[",i,"]=",fmt.Sprintf("0x%02x", data[i]))
	fmt.Print("  ASCII[",i,"]=",fmt.Sprintf("%v", data[i]))
    	if ( data[i] != 144 && data[i] != 152 ){    	
        	fmt.Print("\tChar[",i,"]=",fmt.Sprintf("%c", data[i]),"\n")
        } else {
        	fmt.Print("\tChar[",i,"]= \n")
        }
    }
    fmt.Print("\n\nSTRING: \n\n");
    for i := 0; i < count; i++ {
        if ( data[i] != 0 ){
            if unicode.IsDigit(rune(data[i])) || unicode.IsLetter(rune(data[i])) || unicode.IsSymbol(rune(data[i]))  {
                fmt.Print(fmt.Sprintf("%c", data[i]))
            } else {
                fmt.Print(" ")
            }
        }
    }
    file.Close()
    fmt.Print("\n\nFIN DEL ANALISIS...\n\n");
}

